File: Java-Core/src/main/java/presto/literal/DateLiteral.java
package presto.literal;

import org.joda.time.LocalDate;

import presto.runtime.Context;
import presto.type.DateType;
import presto.type.IType;
import presto.value.Date;


public class DateLiteral extends Literal<Date> {

	public DateLiteral(String text) {
		super(text,parseDate(text.substring(1,text.length()-1)));
	}
	
	@Override
	public IType check(Context context) {
		return DateType.instance();
	}
	
	public static Date parseDate(String text) {
		return new Date(LocalDate.parse(text));
	}
	
	
	
	
}

File: Java-CodeServer/src/main/resources/js/prompto/literal/DateLiteral.js
var Literal = require("./Literal").Literal;
var DateType = require("../type/DateType").DateType;
var LocalDate = require("../value/LocalDate").LocalDate;

function DateLiteral(text) {
	Literal.call(this, text, LocalDate.Parse(text.substring(1,text.length-1)));
	return this;
}

DateLiteral.prototype = Object.create(Literal.prototype);
DateLiteral.prototype.constructor = DateLiteral;

DateLiteral.prototype.check = function(context) {
	return DateType.instance;
};

exports.DateLiteral = DateLiteral;
